### PR TITLE
libcheck: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
TODO
- Requires #18644 to be merged first.

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Comment by valgur** [comment]
The subunit dependency was temporarily disabled with this commit: https://github.com/conan-io/conan-center-index/pull/18793/commits/30cf12758ecab3372318c89bd556ab72d1b797ce
Should revert it once the that PR gets merged.

**Comment by RubenRBS** [comment]
Oh, great to know, will focus on that one first then, thanks for the info!

File: recipes/libcheck/all/conanfile.py [hunk 1]
```diff
@@ -24,96 +29,79 @@ class LibCheckConan(ConanFile):
...
             self.cpp_info.components["liblibcheck"].requires.append("subunit::libsubunit")
         if not self.options.shared:
             if self.settings.os in ["Linux", "FreeBSD"]:
                 self.cpp_info.components["liblibcheck"].system_libs = ["m", "pthread", "rt"]
 
+        # TODO: to remove in conan v2
         bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bin_path))
+        self.output.info(f"Appending PATH environment variable: {bin_path}")
         self.env_info.PATH.append(bin_path)
```
Comment (jwillikers) [inline recipes/libcheck/all/conanfile.py:97]: You can drop this print statement and I believe you can even drop the following line as it should be added to `PATH` automatically.

```suggestion

```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/libcheck/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/libcheck/all/conanfile.py:97]: Oh, really? Does that work with Conan v1 as well? I could have deleted so much dead code. :grin: 
Do you have an example of a recipe where `test_v1_package` works without one, maybe?

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**